tests/proofs/vote-spec.k

```diff
@@ -1,85 +1,86 @@
-requires "lemmas.md"
-
 module VERIFICATION
-  imports LEMMAS
+  imports MICHELSON
 endmodule
 
+
 module VOTE-SPEC
   imports VERIFICATION
 
-  rule <k> AMOUNT .AnnotationList ;
-           PUSH .AnnotationList mutez .AnnotationList 5000000 ;
-           COMPARE .AnnotationList ;
-           GT .AnnotationList ;
-           IF .AnnotationList { FAIL .AnnotationList } { } ;
-           DUP .AnnotationList ; DIP .AnnotationList { CDR .AnnotationList ; DUP .AnnotationList } ;
-           CAR .AnnotationList ; DUP .AnnotationList ;
-           DIP .AnnotationList {
-               GET .AnnotationList ;
-               ASSERT_SOME .AnnotationList ;
-               PUSH .AnnotationList int .AnnotationList 1 ;
-               ADD .AnnotationList ;
-               SOME .AnnotationList
-           } ;
-           UPDATE .AnnotationList ;
-           NIL .AnnotationList operation .AnnotationList ;
-           PAIR .AnnotationList => .K
-        </k>
-        <stack> [(pair string (map string int)) Pair S:String M:Map ] ; .Stack
-             => [(pair (list operation) (map string int)) Pair .InternalList  M[S <- (?I +Int 1)] ]  ; .Stack </stack>
-        <myamount> #Mutez(A) </myamount>
+  claim <k> AMOUNT .AnnotationList ;
+            PUSH .AnnotationList mutez .AnnotationList 5000000 ;
+            COMPARE .AnnotationList ;
+            GT .AnnotationList ;
+            IF .AnnotationList { FAIL .AnnotationList } { } ;
+            DUP .AnnotationList ; DIP .AnnotationList { CDR .AnnotationList ; DUP .AnnotationList } ;
+            CAR .AnnotationList ; DUP .AnnotationList ;
+            DIP .AnnotationList {
+                GET .AnnotationList ;
+                ASSERT_SOME .AnnotationList ;
+                PUSH .AnnotationList int .AnnotationList 1 ;
+                ADD .AnnotationList ;
+                SOME .AnnotationList
+            } ;
+            UPDATE .AnnotationList ;
+            NIL .AnnotationList operation .AnnotationList ;
+            PAIR .AnnotationList
+         => .K
+         </k>
+         <stack> [ (pair string (map string int)) Pair S:String M:Map ] ; .Stack
+              => [ (pair (list operation) (map string int)) Pair .InternalList M[S <- ?I:Int +Int 1] ] ; .Stack </stack>
+         <myamount> #Mutez(A) </myamount>
     requires A >Int 5000000
      andBool A <Int 9223372036854775808
-     andBool M[S] ==K ?I:Int
-     andBool #lookup( M, S, int) ==K Some ?I:Int
+     andBool S in_keys(M)
+    ensures  ?I ==K M[S]
```

Comment:
This encoding tripped me up for a second:

-   matching the updated value in the map to `?I +int 1` to record the value stored in the update) and;
-   asserting that `?I` is equal to the expected value.

It's clever, and I'm really pleased that the backend is happy with it.
Thanks.